src/main/java/com/evanlennick/retry4j/AsyncCallExecutor.java

```diff
@@ -75,31 +103,60 @@ private void executeFuture(Callable<T> callable, String callName, CallExecutor<T
         }
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> afterFailedTry(RetryListener<T> listener) {
         this.afterFailedTryListener = listener;
         return this;
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> beforeNextTry(RetryListener<T> listener) {
         this.beforeNextTryListener = listener;
         return this;
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> onCompletion(RetryListener<T> listener) {
         this.onCompletionListener = listener;
         return this;
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> onSuccess(RetryListener<T> listener) {
         this.onSuccessListener = listener;
         return this;
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> onFailure(RetryListener<T> listener) {
         this.onFailureListener = listener;
         return this;
     }
 
+    public RetryConfig getConfig() {
+        return config;
+    }
+
+    public RetryListener<T> getAfterFailedTryListener() {
```

Comment:
in a later issue, we could pull up many of these getters to interface to avoid duplication between CallExecutor and AsyncCallExecutor